torchaudio/datasets/librilight_limited.py

```diff
@@ -0,0 +1,99 @@
+import os
+from pathlib import Path
+from typing import Tuple, Union
+
+import torchaudio
+from torch import Tensor
+from torch.hub import download_url_to_file
+from torch.utils.data import Dataset
+from torchaudio.datasets.utils import (
+    extract_archive,
+)
+
+
+_URL = "https://dl.fbaipublicfiles.com/librilight/data/librispeech_finetuning.tgz"
+_CHECKSUM = "5d1efdc777b548194d7e09ba89126e2188026df9fd57aa57eb14408d2b2342af"
+
+
+def _get_files(path, split, _ext_audio):
+    if split == "10min":
+        files = sorted(str(p) for p in Path(path).glob("1h/0/*/*/*/*" + _ext_audio))
+    elif split in ["1h", "10h"]:
+        files = [str(p) for p in Path(path).glob("1h/*/*/*/*/*" + _ext_audio)]
+        if split == "10h":
+            files += [str(p) for p in Path(path).glob("9h/*/*/*/*" + _ext_audio)]
+        files = sorted(files)
+    else:
+        raise ValueError(f"Unsupported split value. Found {split}.")
+    return files
+
+
+def _load_item(file_path: str, ext_audio: str, ext_txt: str) -> Tuple[Tensor, int, str, int, int, int]:
```

Comment:
thoughts on importing and reusing `load_librispeech_item` from the librispeech dataset instead of reimplementing the same logic?

Reply:
The difference is that `LibriLightLimited` contains parent folders `1h` and `9h`, which is not compatible with the file path in `load_librispeech_item`. I can refactor `load_librispeech_item` to fit both datasets, but that may make viewing the code inconvenient.

Reply:
The structure of `LibriSpeech` dataset is like `subset/speaker_id/chapter_id/file.flac`, while `LibriLightLimited` dataset looks like `1h/[0-5]/[clean, other]/speaker_id/chapter_id/file.flac` for 1 hour directory, and `9h/[clean, other]/speaker_id/chapter_id/file.flac` for 9 hour directory.

Reply:
sure, in LibriSpeech's `load_librispeech_item`, `path` is the directory of the desired subset, for example, `/root/dev-clean/`, `fileid` is the filename with format `speaker_id-chapter_id-utterance_id.flac`, with that we can find the location of transcription file, which is `speaker_id-chapter_id.trans.txt'.

In LibriLightLimited, the `path` is not general to every subset. `10min` subset uses `/root/1h/0/...` directory, while `1h` subset uses `/root/1h/`, and `10h` subset uses both `/root/1h/` and `/root/9h/`. And because of the `clean` and `other` subdirectories and `[0-5]` subdirectories in `1h`, we can't reconstruct the path of transcription based on `fileid`.

I can refactor LibriSpeech class so that they can use the same method here, what do you think? Notice if we do that, we need to change the usage in librispeech recipes, like https://github.com/pytorch/audio/blob/main/examples/asr/emformer_rnnt/librispeech/lightning.py#L32

Reply:
got it, i think i understand. so, i suppose if we want to use `load_librispeech_item` as is in `__getitem__`, it seems like we could extract the fileid and the dirname preceding the speaker id and chapter id from `file_path`, and then call `load_librispeech_item` on those. would that work?

Reply:
I think so, the `path` argument needs to be tuned per `fileid`, not like in LibriSpeech which can use `self._path` for every file.